packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityMethodChannelHandler.java

```diff
@@ -0,0 +1,163 @@
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+
+public class ConnectivityMethodChannelHandler
+    implements MethodChannel.MethodCallHandler, EventChannel.StreamHandler {
+
+  private final Context context;
+  private ConnectivityManager manager;
+  private BroadcastReceiver receiver;
+
+  public ConnectivityMethodChannelHandler(Context context) {
+    assert (context != null);
+    this.context = context;
+    this.manager =
+        (ConnectivityManager)
+            context.getApplicationContext().getSystemService(Context.CONNECTIVITY_SERVICE);
+  }
+
+  @Override
+  public void onListen(Object arguments, EventChannel.EventSink events) {
+    context.registerReceiver(receiver, new IntentFilter(ConnectivityManager.CONNECTIVITY_ACTION));
+  }
+
+  @Override
+  public void onCancel(Object arguments) {
+    context.unregisterReceiver(receiver);
+    receiver = null;
+  }
+
+  @Override
+  public void onMethodCall(MethodCall call, MethodChannel.Result result) {
+    switch (call.method) {
+      case "check":
+        handleCheck(call, result);
+        break;
+      case "wifiName":
+        handleWifiName(call, result);
+        break;
+      case "wifiBSSID":
+        handleBSSID(call, result);
+        break;
+      case "wifiIPAddress":
+        handleWifiIPAddress(call, result);
+        break;
+      default:
+        result.notImplemented();
+        break;
+    }
+  }
+
+  private String getNetworkType(ConnectivityManager manager) {
+    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      Network network = manager.getActiveNetwork();
+      NetworkCapabilities capabilities = manager.getNetworkCapabilities(network);
+      if (capabilities == null) {
+        return "none";
+      }
+      if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
+          || capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
+        return "wifi";
+      }
+      if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
+        return "mobile";
+      }
+    }
+
+    return getNetworkTypeLegacy(manager);
+  }
+
+  @SuppressWarnings("deprecation")
+  private String getNetworkTypeLegacy(ConnectivityManager manager) {
+    // handle type for Android versions less than Android 9
+    NetworkInfo info = manager.getActiveNetworkInfo();
+    if (info == null || !info.isConnected()) {
+      return "none";
+    }
+    int type = info.getType();
+    switch (type) {
+      case ConnectivityManager.TYPE_ETHERNET:
+      case ConnectivityManager.TYPE_WIFI:
+      case ConnectivityManager.TYPE_WIMAX:
+        return "wifi";
+      case ConnectivityManager.TYPE_MOBILE:
+      case ConnectivityManager.TYPE_MOBILE_DUN:
+      case ConnectivityManager.TYPE_MOBILE_HIPRI:
+        return "mobile";
+      default:
+        return "none";
+    }
+  }
+
+  private BroadcastReceiver createReceiver(final EventChannel.EventSink events) {
```

Comment:
Recommendation: If I were migrating this, I would probably restructure such that this `BroadcastReceiver` is separated from this `MethodCallHandler` and then bring them together in a third class.